test/Authentication/OAuth2AdapterTest.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * @license   http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ * @copyright Copyright (c) 2014 Zend Technologies USA Inc. (http://www.zend.com)
+ */
+
+namespace ZFTest\MvcAuth\Authentication;
+
+use OAuth2\Request as OAuth2Request;
+use PHPUnit_Framework_TestCase as TestCase;
+use Zend\Http\PhpEnvironment\Request as HttpRequest;
+use Zend\Http\Response as HttpResponse;
+use ZF\MvcAuth\Authentication\OAuth2Adapter;
+
+class OAuth2AdapterTest extends TestCase
+{
+    public function setUp()
+    {
+        $this->oauthServer = $this->getMock('OAuth2\Server');
+        $this->adapter = new OAuth2Adapter($this->oauthServer);
+    }
+
+    /**
+     * @group 83
+     */
+    public function testReturns401ResponseWhenUnableToValidate()
```

Comment:
This test ensures that the OAuth2 authentication adapter returns a 401 response for invalid/missing credentials.